src/state.ts

```diff
@@ -72,19 +72,30 @@ export async function deleteWorkloadAlreadyScanned(
 
 export async function getWorkloadImageAlreadyScanned(
   workload: WorkloadAlreadyScanned,
+  imageName: string,
   imageId: string,
 ): Promise<string | undefined> {
-  const key = getWorkloadImageAlreadyScannedKey(workload, imageId);
-  return state.imagesAlreadyScanned.get(key);
+  const key = getWorkloadImageAlreadyScannedKey(workload, imageName);
+  const hasImageId = state.imagesAlreadyScanned.get(key)?.has(imageId);
```

Comment:
I think this will still cause the race condition? what happens if `sha:xxx-old-image` exists in the cache then `sha:yyy-new-image` comes along and replaces it but then another workload with `sha:xxx-old-image` comes along and overwrites it again?

Reply:
maybe we should keep a record of which shas we have seen per image name and just scan and send if it has not been seen before?

Reply:
I think the `imagesLruCacheOptions` becoming a `Set` should prevent the overwriting, only the not existing image Id would be able to add to images (in L93)?

Reply:
>but then another workload

That gets a new key in the cache, so it won't overwrite the image 🙏 And now that we have a Set<>, we can track all occurrences of a name:tag+image-sha for a workload

Reply:
aha great! sorry, I misunderstood! 🤩